SPEC.md

```diff
@@ -106,10 +72,29 @@ Errors are indicated by a non-zero return code and the following JSON being prin
 ```
 
 Error codes 0-99 are reserved for well-known errors (to be defined later).
-Values of 100+ can be freely used for plugin specific errors. 
+Values of 100+ can be freely used for plugin specific errors.
 
 In addition, stderr can be used for unstructured output such as logs.
 
+### DEL command
+
+The DEL command is executed when the container needs to be deleted (leave) from a network.
+It is typically done as part of the container teardown.
+
+#### Parameters passed via environment variables
+- `CNI_COMMAND`: `DEL`
+- `CNI_CONTAINERID`: Container ID. This must be the same Container ID that was passed in during the ADD command. If no Container ID was passed into the ADD command, no ID must be passed in during the DEL command.
+- `CNI_NETNS`: Path to network namespace file representing network namespace from which the container is to be deleted, i.e. `/proc/[pid]/ns/net` or a bind-mount/link to it.
+- `CNI_PATH`: Colon-separated list of paths to search for CNI plugin executables.
+
+#### State streamed via standard input
+This is the JSON value that was returned in the `state` field of the successful result JSON document during the ADD command.
```

Comment:
Hmm. My interpretation is a little different, from the RFC:
https://tools.ietf.org/html/rfc4627#section-2


On Wed, Nov 18, 2015 at 3:57 PM, Eugene Yakubovich <notifications@github.com
> wrote:

> In SPEC.md <https://github.com/appc/cni/pull/79#discussion_r45280495>:
>
> >
> >  In addition, stderr can be used for unstructured output such as logs.
> >
> > +### DEL command
> > +
> > +The DEL command is executed when the container needs to be deleted (leave) from a network.
> > +It is typically done as part of the container teardown.
> > +
> > +#### Parameters passed via environment variables
> > +- `CNI_COMMAND`: `DEL`
> > +- `CNI_CONTAINERID`: Container ID. This must be the same Container ID that was passed in during the ADD command. If no Container ID was passed into the ADD command, no ID must be passed in during the DEL command.
> > +- `CNI_NETNS`: Path to network namespace file representing network namespace from which the container is to be deleted, i.e. `/proc/[pid]/ns/net` or a bind-mount/link to it.
> > +- `CNI_PATH`: Colon-separated list of paths to search for CNI plugin executables.
> > +
> > +#### State streamed via standard input
> > +This is the JSON value that was returned in the `state` field of the successful result JSON document during the ADD command.
>
> JSON document/text can be an object, array, number, string, bool, or null (
> http://www.ecma-international.org/publications/files/ECMA-ST/ECMA-404.pdf).
> So any value that was in state field is itself a valid JSON doc.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/appc/cni/pull/79/files#r45280495>.
>